packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/semanticdata/services/SemanticDataSearchService.java

```diff
@@ -39,7 +40,12 @@ public SemanticDataSearchService(ISemanticDataRepository semanticDataRepository)
 
     @Override
     public Optional<SemanticData> findByProject(AggregateReference<Project, String> project) {
-        return this.semanticDataRepository.findByProjectId(project.getId());
+        return this.semanticDataRepository.findByProjectId(project.getId()).stream().findFirst();
```

Comment:
I don't think we will need to change that but I think that you are missing some concept to capture which semantic data to use for a given project, we need to find out how to keep this information in the database.